internal/controller/pdbwatcher_controller.go

```diff
@@ -102,14 +122,52 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	deployment := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, err // Error fetching Deployment
 	}
 
+	// Check if the resource version has changed
+	if pdbWatcher.Status.ResourceVersion != "" && pdbWatcher.Status.ResourceVersion != deployment.ResourceVersion {
+		// Update ResourceVersion and MinReplicas
+		pdbWatcher.Status.ResourceVersion = deployment.ResourceVersion
+		pdbWatcher.Status.MinReplicas = *deployment.Spec.Replicas
```

Comment:
This is probably another example of how you probably don't actually need maxreplicas in your status as you can always calculate from deployment. Status shoudl be either for things you need to persist things user might want to see for informational purposes.